index.js

```diff
@@ -0,0 +1,327 @@
+        const userData = config.params.data;
+        const table = this.tables[config.table];
+        const model = this.models[config.table];
+
+        if (!table) {
+            return Promise.reject(new Error(`Invalid table name "${config.table}"`));
+        }
+
+        userData.id = id;
+
+        return encodeToDialect(this.client.getDialect(), userData, model)
+            .then(item => table.create(item))
+            .then(() => userData);
+    }
+
+    /**
+     * Remove an item from the table by primary key
+     * @param  {Object}   config             Configuration object
+     * @param  {String}   config.table       Name of the table to interact with
+     * @param  {Object}   config.params      Record Data
+     * @param  {String}   config.params.id   ID of the entry to remove
+     * @return {Promise}                     Resolves to null if remove successfully
+     */
+    _remove(config) {
+        const table = this.tables[config.table];
+
+        if (!table) {
+            return Promise.reject(new Error(`Invalid table name "${config.table}"`));
+        }
+
+        return table.destroy({
+            where: {
+                id: config.params.id
+            }
+        }).then(() => null);
+    }
+
+    /**
+     * Update a record in the datastore
+     * @param  {Object}   config             Configuration object
+     * @param  {String}   config.table       Table name
+     * @param  {Object}   config.params      Record data
+     * @param  {String}   config.params.id   Unique id. Typically the desired primary key
+     * @param  {Object}   config.params.data The data to update with
+     * @return {Promise}                     Resolves to the record that was updated
+     */
+    _update(config) {
+        const id = config.params.id;
+        const userData = config.params.data;
+        const table = this.tables[config.table];
+        const model = this.models[config.table];
+
+        if (!table) {
+            return Promise.reject(new Error(`Invalid table name "${config.table}"`));
+        }
+
+        userData.id = id;
+
+        return encodeToDialect(this.client.getDialect(), userData, model)
+            .then(item => table.update(item, {
+                where: { id }
+            }))
+            .then(() => userData);
+    }
+
+    /**
+     * Scan records in the datastore
+     * @method scan
+     * @param  {Object}   config                Configuration object
+     * @param  {String}   config.table          Table name
+     * @param  {Object}   [config.params]       index => values to query on
+     * @param  {String}   [config.sort]         Sorting option based on GSI range key. Ascending or descending.
+     * @return {Promise}                        Resolves to an array of records
+     */
+    _scan(config) {
+        const table = this.tables[config.table];
+        const model = this.models[config.table];
+        const findParams = {
+            where: {},
+            order: [['id', 'DESC']]
```

Comment:
In dynamo, when we call scanner.sort, it will only sort by GSI range key. That's why we had to define it in data-schema: https://github.com/screwdriver-cd/data-schema/blob/master/models/build.js#L189